### PR TITLE
feat(manager/asdf): support channel suffix on asdf-flutter

### PR DESCRIPTION
**Pull request by ronnnnn**
## Changes

Support channel suffix on asdf-flutter.

## Context

asdf-flutter manages versions of flutter with channel suffix like below.

```shell
$ asdf list-all flutter
...
3.7.12-stable
3.10.0-1.4.pre-beta
3.10.0-1.5.pre-beta
3.10.0-stable
3.11.0-0.0.pre-beta
```

We can install flutter without channel via asdf-flutter.
However, there is a issue that behavior of installing specific version without channel is unstable.
https://github.com/oae/asdf-flutter/pull/43

Therefore, I want to support upgrading with channel suffix.

Related PR: #20786

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/asdf/upgradeable-tooling.ts [hunk 1]
```diff
@@ -144,9 +144,11 @@ export const upgradeableTooling: Record<string, ToolingDefinition> = {
   },
   flutter: {
     asdfPluginUrl: 'https://github.com/oae/asdf-flutter',
-    config: {
+    config: (version) => ({
       datasource: FlutterVersionDatasource.id,
-    },
+      // asdf-flutter plugin supports channel on version suffix.
+      currentValue: version.replace(/-(stable|beta|dev)$/, ''),
```
Comment (viceice) [inline lib/modules/manager/asdf/upgradeable-tooling.ts:150]:
```suggestion
      currentValue: version.replace(regEx(/-(stable|beta|dev)$/), ''),
```

File: lib/modules/manager/asdf/upgradeable-tooling.ts [hunk 2]
```diff
@@ -147,7 +147,7 @@ export const upgradeableTooling: Record<string, ToolingDefinition> = {
     config: (version) => ({
       datasource: FlutterVersionDatasource.id,
       // asdf-flutter plugin supports channel on version suffix.
-      currentValue: version.replace(/-(stable|beta|dev)$/, ''),
+      currentValue: version.replace(regEx(/-(stable|beta|dev)$/), ''),
```
Comment (viceice) [inline lib/modules/manager/asdf/upgradeable-tooling.ts:151]: you need to import from `utils/regex`

Comment (ronnnnn) [reply lib/modules/manager/asdf/upgradeable-tooling.ts:151]: Yes, I fixed it on this.

https://github.com/renovatebot/renovate/commit/6a8abfb0f0e13cc9996eccb119c6741b1c6a4add

**Review by viceice — COMMENTED** [review]
please test on a real repo

**Comment by ronnnnn** [comment]
I tested on [my repo](https://github.com/ronnnnn/renovate-test/pull/2).

**Review by viceice — APPROVED** [review]

**Pull request merged**